src/ember-app.js

```diff
@@ -176,6 +176,24 @@ class EmberApp {
     });
   }
 
+  destroyAppInstance(result, timeout) {
+    const instance = result.instance;
+
+    if (!timeout) return () => instance.destroy();
+
+    // start a timer to destroy the appInstance forcefully in the given ms.
+    // This is a failure mechanism so that node process doesn't get wedged if the `visit` never completes.
+    const destructionTimer = setTimeout(function() {
+      instance.destroy();
+      result.error = new Error(`App instance was forcefully destroyed in ${timeout}ms`);
+    }, timeout);
+
+    return function() {
+      clearTimeout(destructionTimer);
+      instance.destroy();
```

Comment:
We need to make sure we call `instance.destroy()` only once. Can we make sure if the instance is destroyed with a timer doesn't call `instance.destroy()` here?